stdlib/public/core/String.swift

```diff
@@ -594,10 +594,10 @@ extension String {
   /// - Returns: The return value, if any, of the `body` closure parameter.
   @inlinable
   @inline(__always) // Eliminate dynamic type check when possible
-  public func withCString<Result, TargetEncoding: Unicode.Encoding>(
+  public func withCString<Result, Error, TargetEncoding: Unicode.Encoding>(
```

Comment:
How do we want to handle spelled out names vs just E for stdlib API?

Reply:
As long as it doesn't confuse anything I'd favor the spelled-out `Error`

Reply:
I think I'm gonna go with Philippe's version, just because shadowing an existing type name feels like something that will eventually bite someone